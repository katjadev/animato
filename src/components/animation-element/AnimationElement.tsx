import { FC } from 'react'
import { AnimationGroup } from '@animato/types'
import { useEditorState } from '@animato/context/EditorContext/EditorContextProvider'
import Icon from '@animato/components/icon/Icon'
import styles from './AnimationElement.module.css'

export type AnimationElementTranslations = {
  expand: string;
  collapse: string;
}

interface AnimationElementProps {
  element: AnimationGroup;
  translations: AnimationElementTranslations;
}

const AnimationElement: FC<AnimationElementProps> = ({ element, translations }) => {
  const { state: { collapsedAnimations, selectedElementIds }, actions } = useEditorState()

  const collapsed = collapsedAnimations.includes(element.id)
  const selected = selectedElementIds.includes(element.id)

  const toggleCollapsed = () => {
    if (collapsed) {
      actions.expandAnimation({ id: element.id })
    } else {
      actions.collapseAnimation({ id: element.id })
    }
  }

  return (
    <>
      <div className={`${styles.element} ${selected ? styles.selected : ''}`}>
        <div>{element.title}</div>
        <button
          className={styles.collapseButton}
          aria-label={collapsed ? translations.expand : translations.collapse}
          onClick={toggleCollapsed}
        >
          {collapsed ? <Icon icon='nav-arrow-right' /> : <Icon icon='nav-arrow-down' />}
        </button>
      </div>
      {!collapsed && (
        <div className={styles.animationList}>
          {element.animations.map((animation) => (
            <div 
              key={animation.id} 
              className={styles.animation}
            >
              {animation.title}
            </div>
          ))}
        </div>
      )}
    </>
  )
}

export default AnimationElement