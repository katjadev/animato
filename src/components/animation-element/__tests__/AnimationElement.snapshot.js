import { render } from '@testing-library/react'
import AnimationElement from '../AnimationElement'

const translations = {
  expand: 'Expand',
  collapse: 'Collapse',
}

const TEST_ID = 'test-id'
const TEST_ELEMENT = {
  id: TEST_ID,
  title: 'Animation Group',
  animations: [
    { id: 1, title: 'Animation 1' },
    { id: 2, title: 'Animation 2' },
  ],
}

describe('AnimationElement', () => {
  it('renders correctly', () => {
    const { container } = render(
      <AnimationElement 
        element={TEST_ELEMENT} 
        selected={false} 
        translations={translations}
      />
    )
    
    expect(container).toMatchSnapshot()
  })

  it('enders correctly when selected', () => {
    const { container } = render(
      <AnimationElement 
        element={TEST_ELEMENT} 
        selected={true} 
        translations={translations}
      />
    )

    expect(container).toMatchSnapshot()
  })
})
