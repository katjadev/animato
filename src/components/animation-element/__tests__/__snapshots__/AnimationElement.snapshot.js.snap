// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimationElement enders correctly when selected 1`] = `
<div>
  <div
    class="element selected"
  >
    <div>
      Animation Group
    </div>
    <button
      aria-label="Collapse"
      class="collapseButton"
    >
      <i
        class="iconoir-nav-arrow-down "
      />
    </button>
  </div>
  <div
    class="animationList"
  >
    <div
      class="animation"
    >
      Animation 1
    </div>
    <div
      class="animation"
    >
      Animation 2
    </div>
  </div>
</div>
`;

exports[`AnimationElement renders correctly 1`] = `
<div>
  <div
    class="element "
  >
    <div>
      Animation Group
    </div>
    <button
      aria-label="Collapse"
      class="collapseButton"
    >
      <i
        class="iconoir-nav-arrow-down "
      />
    </button>
  </div>
  <div
    class="animationList"
  >
    <div
      class="animation"
    >
      Animation 1
    </div>
    <div
      class="animation"
    >
      Animation 2
    </div>
  </div>
</div>
`;
